### ISSUE TITLE
torch.cat(): expected a non-empty list of Tensors

### ISSUE THREAD

**Issue by hjj-lmx** [RVC-Boss/GPT-SoVITS#2495]
def get_bert_feature(self, text: str, word2ph: list) -> torch.Tensor:
        with torch.no_grad():
            inputs = self.tokenizer(text, return_tensors="pt")
            for i in inputs:
                inputs[i] = inputs[i].to(self.device)
            res = self.bert_model(**inputs, output_hidden_states=True)
            res = torch.cat(res["hidden_states"][-3:-2], -1)[0].cpu()[1:-1]
        assert len(word2ph) == len(text)
        phone_level_feature = []
        for i in range(len(word2ph)):
            repeat_feature = res[i].repeat(word2ph[i], 1)
            phone_level_feature.append(repeat_feature)
        phone_level_feature = torch.cat(phone_level_feature, dim=0)
        return phone_level_feature.T
有一些文字这段代码运行到 torch.cat(phone_level_feature, dim=0)时报错了，phone_level_feature好像是一个空值

**Comment by durongzheng** [comment]
在英文推理时，我也曾碰到类似的情况。原因是对于某些特殊的口语单词，如：gonna, gotta.
数据预处理所用的 nltk，和推理所用的 transfomers.AutoTokenizer 分词方式不一样，造成二者的音素无法对齐。

https://github.com/mockvox/mockvox

在GPT-SoVITS基础上做了一些优化。

**Comment by hjj-lmx** [comment]
> 在英文推理时，我也曾听到类似的情况。原因是对于某些特殊的口语单词，例如：gonna, gotta. 数据常用的nltk，和推理所用的transformers.AutoTokenizer分词方式不一样，造成两者的音素无法配对。
> 
> https://github.com/mockvox/mockvox
> 
> 在GPT-SoVITS基础上做了一些优化。

有在线版本的demo测试吗？

**Comment by durongzheng** [comment]
sorry, 目前没有。需要下载.

**Comment by RVC-Boss** [comment]
需要输入参考文本，不输的话就是对空文本提取。或者你输入的是英文，然后语种写的中文，英文就会自动被过滤，就空了。

**Comment by hjj-lmx** [comment]
> 需要输入参考文本，不输的话就是对空文本提取。或者你输入的是英文，然后语种写的中文，英文就会自动被过滤，就空了。

找到是什么问题了

**Comment by hjj-lmx** [comment]
> 需要输入参考文本，不输的话就是对空文本提取。或者你输入的是英文，然后语种写的中文，英文就会自动被过滤，就空了。

还想请教一个问题

        if have_num:
            temp_list = lang_list
            lang_list = []
            for i, temp_item in enumerate(temp_list):
                if temp_item['lang'] == 'digit':
                    if default_lang:
                        temp_item['lang'] = default_lang
                    elif lang_list and i == len(temp_list) - 1:
                        temp_item['lang'] = lang_list[-1]['lang']
                    elif not lang_list and i < len(temp_list) - 1:
                        temp_item['lang'] = temp_list[1]['lang']
                    elif lang_list and i < len(temp_list) - 1:
                        if lang_list[-1]['lang'] == temp_list[i + 1]['lang']:
                            temp_item['lang'] = lang_list[-1]['lang']
                        elif lang_list[-1]['text'][-1] in [",",".","!","?","，","。","！","？"]:
                            temp_item['lang'] = temp_list[i + 1]['lang']
                        elif temp_list[i + 1]['text'][0] in [",",".","!","?","，","。","！","？"]:
                            temp_item['lang'] = lang_list[-1]['lang']
                        elif temp_item['text'][-1] in ["。","."]:
                            temp_item['lang'] = lang_list[-1]['lang']
                        elif len(lang_list[-1]['text']) >= len(temp_list[i + 1]['text']):
                            temp_item['lang'] = lang_list[-1]['lang']
                        else:
                            temp_item['lang'] = temp_list[i + 1]['lang']
                    else:
                        temp_item['lang'] = 'zh'

                lang_list = merge_lang(lang_list,temp_item)
如果我输入的是“前半规管裂修补等颅中窝手术以及在侧颅底手术中的双镜联合技术等在国内外文献中均有报道。在耳郭再造领域、3D耳模在自体肋软骨移植耳郭再造术中的应用日渐成熟，3D耳模可有效辅助精细解剖结构的重建，”。在第二个“3D”位置会变成“three D”,而且第一个“3D”不是连读的，是先读“三”，停电一下后才读“D”
lang_list[-1]['text'][-1] in [",",".","!","?","，","。","！","？"]:
                            temp_item['lang'] = temp_list[i + 1]['lang']
符合了这个判断，这里的判断逻辑是不是还需要优化
